### PR TITLE
treewide: CI, cmake fixes

### PR DESCRIPTION
**Pull request by woodruffw**
Fixes up the CI to include dynamic Windows builds; fixes up the CMake files in a few places to make packaging with vcpkg a little simpler.

### CODE REVIEW

**Comment by woodruffw** [comment]
I'm going to cut this as 1.2 once merged and make a corresponding PR to the vcpkg upstream.

File: pe-parser-library/src/unicode_winapi.cpp [hunk 1]
```diff
@@ -1,18 +1,14 @@
 /*
 The MIT License (MIT)
-
```
Comment (ekilmer) [inline pe-parser-library/src/unicode_winapi.cpp:3]: Did you mean to remove these lines?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -82,19 +86,29 @@ jobs:
...
     - name: build
       run: |
         mkdir build
         cd build
-        cmake -G "Visual Studio 16 2019" -A x64 ..
+        cmake `
+          -G "Visual Studio 16 2019" `
+          -A ${{ matrix.build-arch }} `
+          -DBUILD_SHARED_LIBS=${{ matrix.build-shared }} `
+          ..
```
Comment (ekilmer) [inline .github/workflows/ci.yml:103]: Never knew the ` (back-tick) character was used for multi-line commands in Windows 👀

File: pe-parser-library/src/unicode_winapi.cpp (hunk 1 above)
Comment (woodruffw) [reply pe-parser-library/src/unicode_winapi.cpp:3]: Bleh, that's from a bad revert. Will fix.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (woodruffw) [reply .github/workflows/ci.yml:103]: Yeah, I learned that as part of the uthenticode CI setup 😅

Comment (woodruffw) [reply .github/workflows/ci.yml:103]: (It's a powershell specific thing, apparently).

**Pull request merged**